zerver/openapi/markdown_extension.py

```diff
@@ -515,26 +515,28 @@ def __init__(self, md: markdown.Markdown, config: Mapping[str, Any]) -> None:
         super().__init__(MACRO_REGEXP_DESC, md, config)
 
     def render(self, function: str) -> List[str]:
-        description: List[str] = []
         path, method = function.rsplit(":", 1)
         description_dict = get_openapi_description(path, method)
-        description_dict = description_dict.replace("{{api_url}}", self.api_url)
-        description.extend(description_dict.splitlines())
-        return description
+        return description_dict.splitlines()
 
 
 class APITitlePreprocessor(BasePreprocessor):
     def __init__(self, md: markdown.Markdown, config: Mapping[str, Any]) -> None:
         super().__init__(MACRO_REGEXP_TITLE, md, config)
 
     def render(self, function: str) -> List[str]:
-        title: List[str] = []
         path, method = function.rsplit(":", 1)
         raw_title = get_openapi_summary(path, method)
-        title.extend(raw_title.splitlines())
-        title = ["# " + line for line in title]
+        title = [
+            *("# " + line for line in raw_title.splitlines()),
+            "",
+            f"`{method.upper()} {self.api_url}/v1{path}`",
+        ]
         if check_requires_administrator(path, method):
-            title.append("{!api-admin-only.md!}")
+            title += [
+                "",
+                "{!api-admin-only.md!}",
```

Comment:
These changes look good to me. My one thought is that the documentation looks a bit cleaner when the admin only warning is directly after the title, so maybe move this check before the method and path?

<details>
<summary>Example screenshot as is:</summary>

![Screenshot from 2022-06-21 16-49-46](https://user-images.githubusercontent.com/63245456/174830470-e0a6a738-50c5-4611-82a2-21bc9bd53e68.png)
</details>

<details>
<summary>Example screenshot with admin warning first:</summary>

![Screenshot from 2022-06-21 16-54-29](https://user-images.githubusercontent.com/63245456/174830779-2c723241-5f03-4577-a50e-1debd8e31e37.png)
</details>

Reply:
Yeah, I agree it looks cleaner with the warning first.

Reply:
I feel it’s less consistent, but not strongly enough to argue. Updated.